tendermint/src/rpc/event_listener.rs

```diff
@@ -171,65 +154,36 @@ impl response::Response for TxResultResult {}
 
 /// Tendermint ABCI Events
 #[derive(Serialize, Deserialize, Debug, Clone)]
-struct TmEvent {
+pub struct TmEvent {
     #[serde(rename = "type")]
     event_type: String,
     attributes: Vec<Attribute>,
 }
 /// Event Attributes
 #[derive(Serialize, Deserialize, Debug, Clone)]
-struct Attribute {
+pub struct Attribute {
     key: String,
     value: String,
 }
 
-/// Block Results
-#[derive(Serialize, Deserialize, Debug, Clone)]
-pub struct RPCBlockResult {
-    query: String,
-    data: BlockResultData,
-    events: HashMap<String, Vec<String>>,
-}
-impl response::Response for RPCBlockResult {}
-
-/// Block Results data
-#[derive(Serialize, Deserialize, Debug, Clone)]
-struct BlockResultData {
-    #[serde(rename = "type")]
-    data_type: String,
-    value: BlockValue,
-}
-
 ///Block Value
 #[derive(Serialize, Deserialize, Debug, Clone)]
-struct BlockValue {
+pub struct EventDataNewBlock {
     block: Option<Block>,
+
+    // TODO these should be the same as abci::responses::BeginBlock
+    // and abci::responses::EndBlock
     result_begin_block: Option<ResultBeginBlock>,
     result_end_block: Option<ResultEndBlock>,
 }
 
 /// Begin Block Events
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct ResultBeginBlock {
-    events: Vec<TmEvent>,
+    events: Option<Vec<TmEvent>>,
 }
 ///End Block Events
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct ResultEndBlock {
-    validator_updates: Vec<Option<serde_json::Value>>,
-}
-
-impl RPCTxResult {
-    /// Extract events from TXEvent if event matches are type query
-    pub fn extract_events(
-        &self,
-        action_query: &str,
-    ) -> Result<HashMap<String, Vec<String>>, Box<dyn stdError>> {
-        if let Some(message_action) = self.events.get("message.action") {
-            if message_action.contains(&action_query.to_owned()) {
-                return Ok(self.events.clone());
-            }
-        }
-        Err("Incorrect Event Type".into())
-    }
```

Comment:
This function is required for actually providing types for the events. It can't be removed. Needs to be added to the new ResultEvent type.

Reply:
Thanks, I'll add it back now. Did everything else work with gaia too?

Reply:
Just to confirm my understanding: Above in `get_event` you don't just need the `TMEventData` but the whole `ResultEvent`? This:
```rust
pub struct ResultEvent {
    query: Option<String>,
    data: Option<TMEventData>,
    events: Option<HashMap<String, Vec<String>>>,
}
```
Is that correct?

Reply:
I think it's reasonable for the caller to be interested in both the TMEventData and the Hashmap of Events

Reply:
Changed to return `ResultEvent` in https://github.com/informalsystems/tendermint-rs/pull/279/commits/0e3e422ae713e439a6a2e2cf04c2a9f59a098dcd